iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -140,14 +173,43 @@ private async Task RefreshLoopAsync(DateTime refreshesOn, CancellationToken canc
             }
         }
 
-        public void StopLoop()
+        // StopLoop ensures that the resources related to token refresher are cancelled and ready for cleanup.
+        // It will cancel the associated cancellation token source and will await completion of the refresh loop task.
+        // Any exceptions thrown as a result of these operations are caught and logged.
+        // StopLoop will not generate any exceptions so it safe for us to declare this as async void.
```

Comment:
Still don't do this. It hurts nothing to return a Task.

Reply:
The call chain doesn't prevent this from returning a task. It just prevents the callers from awaiting it.

Also, we can choose to use IAsyncDisposable, if we wish.